src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -171,114 +204,137 @@ private WordEmbeddingsTransform(IHost host, ModelLoadContext ctx, IDataView inpu
             else
             {
                 _modelKind = (PretrainedModelKind)ctx.Reader.ReadUInt32();
-                _modelFileNameWithPath = EnsureModelFile(Host, out _linesToSkip, (PretrainedModelKind)_modelKind);
+                _modelFileNameWithPath = EnsureModelFile(Host, out linesToSkip, (PretrainedModelKind)_modelKind);
             }
 
             Host.CheckNonWhiteSpace(_modelFileNameWithPath, nameof(_modelFileNameWithPath));
-            _currentVocab = GetVocabularyDictionary();
-            _outputType = new VectorType(NumberType.R4, 3 * _currentVocab.Dimension);
-            Metadata.Seal();
+            _currentVocab = GetVocabularyDictionary(linesToSkip);
```

Comment:
>GetVocabularyDictionary [](start = 28, length = 23)

Can we move file reading to mapper?
Since you use trivial estimator you basically create transformer in estimator constructor.
Which would force heavy IO processing, and I think purpose of Estimator to be something light weight.